app/views/contexts/hub_context.py

```diff
@@ -94,17 +91,43 @@ def get_row_for_section(
 
         return context
 
-    def _get_rows(self) -> List[Mapping[str, Union[str, List]]]:
-        rows = []
+    @staticmethod
+    def get_section_url(section_id, list_item_id) -> str:
+        if list_item_id:
+            return url_for(
+                'questionnaire.get_section',
+                section_id=section_id,
+                list_item_id=list_item_id,
+            )
 
-        for section in self._sections:
-            section_status = self._progress_store.get_section_status(section['id'])
+        return url_for('questionnaire.get_section', section_id=section_id)
 
-            section_name = section['title']
-            section_url = self.get_section_url(section['id'])
+    def _get_row_for_section(self, section_name, section_id, list_item_id=None):
+        section_status = self._progress_store.get_section_status(
+            section_id, list_item_id
+        )
 
-            rows.append(
-                self.get_row_for_section(section_name, section_status, section_url)
-            )
+        return self.get_row_context_for_section(
+            section_name, section_status, self.get_section_url(section_id, list_item_id)
+        )
+
+    def _get_rows(self) -> List[Mapping[str, Union[str, List]]]:
```

Comment:
Pretty sure its just me, but I don't really understand what rows are in this context. Looked at the function and it just looks like a dict of stuff. Is this for the frontend table of repeating elements ?

Reply:
Yes this is to create the context for the rows displayed on the hub.

Reply:
Thank you